packages/tailwindcss/src/utilities.ts

```diff
@@ -371,6 +378,47 @@ export function createUtilities(theme: Theme) {
     ])
   }
 
+  function spacingUtility(
+    name: string,
+    themeNamespace: ThemeKey | ThemeKey[],
+    handle: (value: string) => AstNode[] | undefined,
+    {
+      supportsNegative = false,
+      supportsFractions = true,
+      ignoredThemeKeys = [],
+    }: {
+      supportsNegative?: boolean
+      supportsFractions?: boolean
+      ignoredThemeKeys?: ThemeKey[]
+    } = {},
+  ) {
+    utilities.static(`${name}-px`, (candidate) => {
+      if (!supportsNegative && candidate.negative) return
+      return handle(candidate.negative ? '-1px' : '1px')
+    })
+    functionalUtility(name, {
+      themeKeys: ([] as ThemeKey[]).concat(themeNamespace, '--spacing'),
+      ignoredThemeKeys,
+      supportsFractions,
+      supportsNegative,
+      handleBareValue: ({ value }) => {
+        let multiplier = theme.resolve(null, ['--spacing'])
+        if (!multiplier) return null
+
+        let num = Number(value)
+        if (num < 0 || num % 0.25 !== 0 || String(num) !== value) return null
+
+        return `calc(${multiplier} * ${value})`
```

Comment:
If the `--spacing` flag has the inline flag, we could avoid the `calc()` here since the value would be static anyways.

Reply:
Lightning CSS will evaluate and minify this for us if the value is inline so I figured we would just not worry about that in our codebase and let Lightning handle it 🙌🏻

Reply:
gotcha!